lib/puppet/metatype/manager.rb

```diff
@@ -146,16 +156,19 @@ def type(name)
     # We are overwhelmingly symbols here, which usually match, so it is worth
     # having this special-case to return quickly.  Like, 25K symbols vs. 300
     # strings in this method. --daniel 2012-07-17
-    return @types[name] if @types[name]
+    return @types[name] if @types.include? name
```

Comment:
Looking at the implementation of `hash.c` provides this:

```c
    rb_define_singleton_method(envtbl,"include?", env_has_key, 1);
    rb_define_singleton_method(envtbl,"member?", env_has_key, 1);
    rb_define_singleton_method(envtbl,"has_key?", env_has_key, 1);
    rb_define_singleton_method(envtbl,"key?", env_has_key, 1);
```

`#include?`, `#member?`, `#has_key?`, and `#key?` are all synonymous.